core/src/main/java/org/datasyslab/geospark/spatialRDD/SpatialRDD.java

```diff
@@ -388,24 +385,52 @@ public void buildIndex(final IndexType indexType,boolean buildIndexOnSpatialPart
 	        }
 	}
 
-    /**
-     * Boundary.
-     *
-     * @return the envelope
-     */
-    public Envelope boundary() {
-    	Object minXEnvelope = this.rawSpatialRDD.min(new XMinComparator());
-    	Object minYEnvelope = this.rawSpatialRDD.min(new YMinComparator());
-    	Object maxXEnvelope = this.rawSpatialRDD.max(new XMaxComparator());
-    	Object maxYEnvelope = this.rawSpatialRDD.max(new YMaxComparator());
-    	Double[] boundary = new Double[4];
-    	boundary[0] = ((Geometry) minXEnvelope).getEnvelopeInternal().getMinX();
-    	boundary[1] = ((Geometry) minYEnvelope).getEnvelopeInternal().getMinY();
-    	boundary[2] = ((Geometry) maxXEnvelope).getEnvelopeInternal().getMaxX();
-    	boundary[3] = ((Geometry) maxYEnvelope).getEnvelopeInternal().getMaxY();
-        this.boundaryEnvelope =  new Envelope(boundary[0],boundary[2],boundary[1],boundary[3]);
-        return this.boundaryEnvelope;
-    }
+	public static final class BoundaryAggregation {
+		public static Envelope combine(Envelope agg1, Envelope agg2) throws Exception {
+			if (agg1 == null) {
+				return agg2;
+			}
+
+			if (agg2 == null) {
+				return agg1;
+			}
+
+			return new Envelope(
+				Math.min(agg1.getMinX(), agg2.getMinX()),
+				Math.max(agg1.getMaxX(), agg2.getMaxX()),
+				Math.min(agg1.getMinY(), agg2.getMinY()),
+				Math.max(agg1.getMaxY(), agg2.getMaxY()));
+		}
+
+		public static Envelope add(Envelope agg, Geometry object) throws Exception {
+			return combine(object.getEnvelopeInternal(), agg);
+		}
+	}
+
+	/**
+	 * Boundary.
+	 *
+	 * @return the envelope
+	 */
+	public Envelope boundary() {
```

Comment:
Use Aggregate to get the boundary is a good solution! But @mbasmanova could you please provide a unit test for this boundary aggregate function? Maybe you can use PointRDD or PolygonRDD to test its boundary.